raw_bsd.go

```diff
@@ -111,13 +117,38 @@ func listenPacket(ifi *net.Interface, proto Protocol) (*packetConn, error) {
 
 // ReadFrom implements the net.PacketConn.ReadFrom method.
 func (p *packetConn) ReadFrom(b []byte) (int, net.Addr, error) {
+	var timeout time.Duration
+	var hasTimeout bool
+
+	p.timeoutMu.Lock()
+	if !p.rtimeout.IsZero() {
+		timeout = p.rtimeout.Sub(time.Now())
+		hasTimeout = true
+	}
+	p.timeoutMu.Unlock()
+
+	if hasTimeout && timeout < time.Microsecond {
```

Comment:
Is this check necessary? If someone does something like `time.Duration(1)` (which is 1ns) it's usually a programmer error.

Reply:
Cool, can you leave a comment to that effect?

Reply:
done